### DISCUSSION TITLE
how to use less in next.js 13.4.13?

### DISCUSSION THREAD

**Discussion by rb05180518** [vercel/next.js#53775] [Help] [answered]
### Summary

My current version of next.js is 13.4.13. I am very anxious about how to use less. I hope everyone can help me answer, whether it is app routing mode or pages routing mode

**Comment by rb05180518** [comment, marked as answer]
yarn add less less-loader next-with-less
`
/** @type {import('next').NextConfig} */
const withLess = require("next-with-less");
module.exports = withLess({});
`

usage： `index.module.less`

Now, this problem has been resolved

**Comment by Jairwin-L** [comment]
seems useless?
code
```tsx
import style from './index.module.less';

export default function Main() {
  return <div className={style.main}>123</div>;
}
```
config
```js
const withLess = require('next-with-less');

/** @type {import('next').NextConfig} */
const nextConfig = withLess({
  distDir: 'dist',
  reactStrictMode: true,
  experimental: {
    forceSwcTransforms: true,
  },
});

module.exports = nextConfig;
```

```bash
CSS Modules cannot be imported from within node_modules.
Read more: https://nextjs.org/docs/messages/css-modules-npm
Location: src/app/page.tsx
```

**Reply by rb05180518** [reply]
From the error message, it can be seen that you have imported a. css or. less file in a non app file. Please ensure that you have not done so

**Reply by Jairwin-L** [reply]
The index.module.less file is referenced, even if index.less is used, it has no effect
```tsx
import './index.less';

export default function Main() {
  return <div className="main">123</div>;
}
```

**Reply by rb05180518** [reply]
except for _app file, you use .less is not effect in other file.

**Reply by Jairwin-L** [reply]
but the _app file belongs to the page router, I am using the app router now, and shouldn’t each page have a less file

**Reply by rb05180518** [reply]
Okay, I haven't tried using app router before. I'll give it a try tomorrow.

**Reply by Jairwin-L** [reply]
thanks

**Reply by rb05180518** [reply]
The app router does not support less files. I hope this document can help you https://nextjs.org/docs/app/building-your-application/styling/css-modules

**Reply by Jairwin-L** [reply]
Okay

**Reply by RickYangzz** [reply]
> Okay

Did you solve this problem?

**Reply by Jairwin-L** [reply]
> > Okay
> 
> Did you solve this problem?

It can be done without using less.

**Reply by BossBele** [reply]
For now, I have reverted to compiling all my less code accordingly (for my case, it is into one CSS file). Then I import the CSS file into my project without issues.

// In package.json
```
"scripts": {
    "prebuild": "find src/styles/ -name '*.less' -exec lessc {} \\; > src/styles/index.css",
    ...
}
```


The command finds all .less files in `./src/styles/` and compiles them into `index.css`
Learn more: https://lesscss.org/usage/#command-line-usage
